cleverhans/utils_tf.py

```diff
@@ -411,3 +415,29 @@ def clip_eta(eta, ord, eps):
         factor = tf.minimum(1., eps / norm)
         eta = eta * factor
     return eta
+
+
+def tf_reduce_sum(input_tensor, axis=None, keepdims=None,
```

Comment:
Given that the code that this function is looking to support is only included in one tutorial folder script, I would rather modify the tutorial's logic directly at line 79 of the `tutorial_models.py` file you modified below and remove this function from the `utils_tf.py` file (we don't want to artificially extend the lifetime of features deprecated in TF by adding wrappers unless the code affected is directly part of the CleverHans core module).

Reply:
yeah makes sense to me.
few issues here, need your suggestion.

i was working on #397, attacks.py has the same depreciation warnings.
tf 1.4 - expects keep_dims and throws error for keepdims.
tf 1.8 - expects keepdims and throws warning for keep_dims.

are we planning to have long time support for tf 1.4 ?

Reply:
Thanks for pointing this out. In that case, I would recommend following the style adopted in `model_train` https://github.com/tensorflow/cleverhans/blob/master/cleverhans/utils_tf.py#L121

As far as I can tell, there are 6 lines in `attacks.py` that are potentially affected by this change in TF naming convention for keep_dims/keepdims. I would recommend modifying these 6 lines directly into something like:

```
if tf above 1.8 
    use keepdims 
else 
    use keep_dims 
    warn user that CleverHans may not support this version of TF in the future
```

Reply:
sure, i will write a if else condition inplace for ( utils_tf - 1, attacks.py - 6, attacks_tf.py - 6) these locations, we can remove them if and when tf 1.4 is not supported.